spring-cloud-gcp-pubsub/src/main/java/com/google/cloud/spring/pubsub/PubSubAdmin.java

```diff
@@ -54,11 +54,11 @@
 
 	private static final String NO_TOPIC_SPECIFIED = "No topic name was specified.";
 
-	private final String projectId;
+	private String projectId;
 
-	private final TopicAdminClient topicAdminClient;
+	private TopicAdminClient topicAdminClient;
```

Comment:
IMO keeping these `final` is a greater benefit than the small amount of duplicate `Assert.notNull` code in the constructors, which, if I estimate correctly, may just be ensuring the project ID provider is not null and the project ID it provides is not null.

Reply:
Agreed.